AnkiDroid/src/main/java/com/ichi2/anki/jsaddons/NpmUtils.kt

```diff
@@ -28,83 +27,60 @@ object NpmUtils {
      * https://github.com/npm/validate-npm-package-name/blob/main/index.js
      * https://github.com/lassjs/is-valid-npm-name/blob/master/index.js
      */
-    fun validateName(name: String): Boolean {
-        if (name.isBlank()) {
-            Timber.i("Invalid package name: non-empty characters allowed")
-            return false
+    @CheckResult
+    // @OptIn(ExperimentalContracts::class)
+    fun validateName(name: String?): List<String>? {
+        // Compiler fails on this contract.
+        //        contract {
+        //            returnsNotNull() implies (name != null)
+        //        }
+        if (name == null) {
+            return listOf("Name is not set.")
         }
-
-        // first trim it
-        if (name.trim() != name) {
-            Timber.i("Invalid package name: non-empty characters allowed")
-            return false
-        }
-
-        // name can no longer contain more than 214 characters
-        if (name.length > 214) {
-            Timber.i("Invalid package name: name string length less than 214 allowed")
-            return false
+        val errors = mutableListOf<String>()
+        for (invalidChar in "~'!()*") {
```

Comment:
```suggestion
        for (invalidChar in "~'!{[()]}* ") {
```
This will catch spaces in names.